src/reducers/pipelineReducer.js

```diff
@@ -0,0 +1,30 @@
+import { Map, List } from 'immutable';
+import { updateActivityList } from './common';
+import {
+  FLAG_GETTING_PIPELINES,
+  FLAG_GETTING_PIPELINE,
+  GET_PIPELINE_SUCCESS,
+  GET_PIPELINES_SUCCESS,
+  FLAG_CLEARING_PIPELINE,
+} from '../shared/types';
+
+const INITIAL_STATE = Map({
+  activity: List(),
+  current: Map(),
+  pipelines: List(),
+});
+
+export default (state = INITIAL_STATE, { type, payload }) => {
+  switch (type) {
+    case FLAG_GETTING_PIPELINES:
+    case FLAG_GETTING_PIPELINE:
+    case FLAG_CLEARING_PIPELINE:
+      return state.updateIn(['activity'], updateActivityList(payload));
+    case GET_PIPELINE_SUCCESS:
+      return state.setIn(['current'], Map(payload));
+    case GET_PIPELINES_SUCCESS:
+      return state.setIn(['pipelines'], List(payload));
+    default:
```

Comment:
The idea of clearing is to remove the data in `current` in order for not displaying the previous data for a short moment before replacing it with new data. So the case `CLEAR_PIPELINE_SUCCESS` is missing.